include/taskolib/Sequence.h

```diff
@@ -290,10 +290,19 @@ class Sequence
      *
      * If the sequence finished normally, the returned string is empty.
      *
-     * \see execute()
+     * \see get_error_step_index(), set_error_message_and_step_index()
      */
     const std::string& get_error_message() const noexcept { return error_message_; }
 
+    /**
+     * Return the index of the step at which the sequence stopped prematurely.
+     *
+     * If the sequence finished normally, this function returns nullopt.
```

Comment:
People might read this as 'If the sequence does not finish normally this will not return nullopt', which is of course wrong.
In other words I believe users should be pointed to the function they should use to determine if an error occured.
I believe that might be `get_error_message()` which will return an empty string on non-error-runs. But that is not really communicated clearly.